Source/Extensions/Blazorise.DataGrid/DataGridColumn.cs

```diff
@@ -39,26 +41,39 @@ public DataGridColumn()
         valueGetter = new( () => FunctionCompiler.CreateValueGetter<TItem>( Field ) );
         valueSetter = new( () => FunctionCompiler.CreateValueSetter<TItem>( Field ) );
         sortFieldGetter = new( () => FunctionCompiler.CreateValueGetter<TItem>( SortField ) );
+
+        currentSortDirection[DataGridSortMode.Single] = SortDirection;
+        currentSortDirection[DataGridSortMode.Multiple] = SortDirection;
+        currentFilterMethod = FilterMethod;
+
+        Filter?.Subscribe( OnSearchValueChanged );
     }
 
     #endregion
 
     #region Methods
 
-    protected override void OnInitialized()
+    /// <summary>
+    /// Provides a way to generate column specific dependencies outside the Blazor engine.
+    /// </summary>
+    /// <param name="parentDataGrid"></param>
+    /// <param name="serviceProvider"></param>
+    internal void InitializeGeneratedColumn( DataGrid<TItem> parentDataGrid, IServiceProvider serviceProvider )

```

Comment:
Name it `InitializeAutoGeneratedColumn`?

Reply:
hmm I didn't name it "Auto" because it can be a column added by user even. Now users should be able to add the columns programatically. They could not before, there were initializers missing.

Reply:
What do you mean? Isn't the method marked as `internal`?

Reply:
I mean, the method should be invoked whenever a generated column is detected. Here, we are refering to generated column as a column that was created imperatively through code instead of declaretevly through the razor code.

So whenever a user calls `DataGrid.AddColumn` it should end up calling this method also, that's why I wanted it to be a more generic name not tied directly to our `AutoGenerateColumns` feature.